cpp/test/IceGrid/allocation/AllTests.cpp

```diff
@@ -214,68 +211,74 @@ class StressClient : public IceUtil::Thread, public IceUtil::Monitor<IceUtil::Mu
     }
 
     void
-    allocateAndDestroy(const SessionPrx& session)
+    allocateAndDestroy(const shared_ptr<SessionPrx>& session)
     {
         ostringstream os;
-        os << "stress-" << IceUtilInternal::random(3);
-        CallbackPtr asyncCB = new Callback();
-        IceGrid::Callback_Session_allocateObjectByIdPtr cb =
-            IceGrid::newCallback_Session_allocateObjectById(asyncCB, &Callback::response, &Callback::exception);
-        session->begin_allocateObjectById(Ice::stringToIdentity(os.str()), cb);
+        os << "stress-" << (_rd() % 3);
+
+        auto asyncCB = make_shared<Callback>();
+        session->allocateObjectByIdAsync(Ice::stringToIdentity(os.str()),
+                                         [asyncCB](shared_ptr<Ice::ObjectPrx> o) { asyncCB->response(move(o)); },
+                                         [asyncCB](exception_ptr e) { asyncCB->exception(move(e)); });
         session->destroy();
     }
 
     void
-    allocateByTypeAndDestroy(const SessionPrx& session)
+    allocateByTypeAndDestroy(const shared_ptr<SessionPrx>& session)
     {
-        CallbackPtr asyncCB = new Callback();
-        IceGrid::Callback_Session_allocateObjectByTypePtr cb =
-            IceGrid::newCallback_Session_allocateObjectByType(asyncCB, &Callback::response, &Callback::exception);
-        session->begin_allocateObjectByType("::StressTest", cb);
+        auto asyncCB = make_shared<Callback>();
+        session->allocateObjectByTypeAsync("::StressTest",
+                                          [asyncCB](shared_ptr<Ice::ObjectPrx> o) { asyncCB->response(move(o)); },
+                                          [asyncCB](exception_ptr e) { asyncCB->exception(move(e)); });
```

Comment:
I wouldn't move this exception_ptr, since it doesn't have any standard-defined ctors.